Cod3rsGrowth.Web/wwwroot/Webapp/app/carros/AdicionarCarro.controller.js

```diff
@@ -0,0 +1,283 @@
+﻿sap.ui.define([
+    "ui5/carro/app/common/BaseController",
+    "sap/ui/core/routing/History",
+    "sap/ui/model/json/JSONModel",
+    "ui5/carro/model/formatter",
+    "ui5/carro/model/validacao",
+	"sap/m/MessageBox"
+
+], function (BaseController, History, JSONModel, Formatter, validacao, MessageBox) {
+    "use strict";
+
+    let _rota;
+    let idCarro;
+    const modeloCores = "Cores";
+    const parametroNome = "name";
+    const modeloMarcas = "Marcas";
+    const idDoInputFlex = "InputFlex";
+    const idDoInputValor = "InputValor";
+    const idDoInputCor = "SelecionarCor";
+    const idDoInputModelo = "InputModelo";
+    const parametroArgumento = "arguments";
+    const idDoInputMarca = "SelecionarMarca";
+    const rotaEditarCarro = "appEditarCarro";
+    const rotaCriarCarro = "appAdicionarCarro";
+    const rotaListagemCarros = "appListagemCarro";
+    const url= "http://localhost:5071/api/Carros/";
+    const idTituloViewCriarCarro = "TituloEditarCarro";
+    const textoParaAdicionarTituloEdicao = "Editar Carro";
+    const textoParaAdicionarTituloCriacao = "Adicionar Carro";
+    const urlCores = "http://localhost:5071/api/Carros/Cores";
+    const idDoMessageStripErroCriarCarro = "erroAoCriarCarro";
+    const urlMarcas = "http://localhost:5071/api/Carros/Marcas";
+    const idDoMessageStripErroEditarCarro = "erroAoEditarCarro";
+    const idDoMessageStripSucessoCriacao = "sucessoAoCriarCarro";
+    const idDoMessageStripSucessoEditar = "sucessoAoEditarCarro";
+
+    return BaseController.extend("ui5.carro.app.carros.AdicionarCarro", {
+        formatter: Formatter,
+
+        onInit() {
+            this.aoCoincidirRota();
+        },
+
+        aoCoincidirRota() {
+            this.processarEvento(() => {
+                var rota = sap.ui.core.UIComponent.getRouterFor(this);
+                rota.getRoute(rotaCriarCarro).attachMatched(this._carregarEventosCriar, this);
+                rota.getRoute(rotaEditarCarro).attachMatched(this._carregarEventodEditar, this);
+            });
+        },
+
+        _carregarEventodEditar(oEvent) {
+            this._mudarTituloDaViewEdicao();
+            this._removerMessageStrip();
+            this._limparInputs();
+            this._obterCarroPorId(oEvent);
+            this._carregarDescricaoCores();
+            this._carregarDescricaoMarcas();
+            var oRouter = this.getRouter();
+            _rota = oRouter.getRoute(oEvent.getParameter(parametroNome))._oConfig.name;
+        },
+
+        _carregarEventosCriar(oEvent) {
+            this._mudarTituloDaViewCriar();
+            this._limparInputs();
+            this._removerMessageStrip();
+            this._carregarDescricaoCores();
+            this._carregarDescricaoMarcas();
+            var oRouter = this.getRouter();
+            _rota = oRouter.getRoute(oEvent.getParameter(parametroNome))._oConfig.name;
+        },
+
+        _limparInputs() {
```

Comment:
acho que isso pode melhorar
Sera que não é melhor passar um modelo vazio para ele em vez de alterar cada um dos inputs.
this.getView().setModel(modelo_vazio, MODELO);